src/streamlink/plugins/wasd.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+
+log = logging.getLogger(__name__)
+
+class WASD(Plugin):
+	_url_re = re.compile(r"https?://(?:www\.)?wasd\.tv/")
+	_url_channel = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/(\d+)")
+	_url_video = re.compile(r"https?://(?:www\.)?wasd\.tv/channel/\d+/videos/(\d+)")
+	
+	@classmethod
+	def can_handle_url(cls, url):
+		return WASD._url_re.match(url)
+		
+	def _get_streams(self):
+		re_video_id = self._url_video.match(self.url)
+		re_channel_id = self._url_channel.match(self.url)
+		
+		if not re_video_id and not re_channel_id:
+			return log.error("Expected to see url like https://wasd.tv/channel/12345 or https://wasd.tv/channel/12345/videos/67890")
```

Comment:
you don't need this, update `_url_re` instead of using `_url_channel` and `_url_video`

example

```py
>>> f = r'https?://(?:www\.)?wasd\.tv/channel/(?P<channel_id>\d+)(?:/videos/(?P<video_id>\d+))?'
>>> import re
>>> r = re.compile(f)
>>> url1 = 'https://wasd.tv/channel/12345/videos/67890'
>>> url2 = 'https://wasd.tv/channel/12345'
>>> r.match(url1).groups()
('12345', '67890')
>>> r.match(url2).groups()
('12345', None)
>>> r.match(url2).group('video_id')
>>> r.match(url2).group('channel_id')
'12345'
```